### DISCUSSION TITLE
Module Federation JavaScript libraries isolation

### DISCUSSION THREAD

**Discussion by nvladimirovi** [webpack/webpack#15283] [General]
In case two applications are using the same library but different versions this may pollute the global scope of the application and therefore one application may break the other. Is there a way to isolate the JavaScript scope of each individual application, in such a way that if application A's lib writes something in the window object application B's lib won't have access to it?

I don't have a concrete example but in general, it may happen.

**Comment by vankop** [comment]
maybe if both apps will use different [global object](https://webpack.js.org/configuration/output/#outputglobalobject), otherwise they have same runtime..

**Reply by nvladimirovi** [reply]
Yes, this solves a big part of the problem, but here is another example, imagine js lib that adds attributes to the body tag and then uses them to perform something, in that case, the 2nd loaded library will get the wrong config if the attribute names conflict.

There are also issues with web components like you can not register web components with the same tag name twice, there is nothing much to do, except maybe rename them.

**Reply by vankop** [reply]
No this is not possible now and I think it will be limitation in future as well..
